W-16047302-Add-default-debugger-settings-CP (#537)
mulesoft/docs-studio: modules/ROOT/pages/visual-debugger-concept.adoc

include::_attributes.adoc[]
endif::[]

> With Anypoint Studio debugger, you can run your application from Studio in debug mode and configure breakpoints to stop the execution of your application at a specific message processor. In the *Mule Debugger* view, inspect the content of the message as it exists at that point in the flow and evaluate a DataWeave expression against it.

Reviewer: ```suggestion
With Anypoint Studio debugger, you can run your application from Studio in debug mode and configure breakpoints to stop application execution at a specific message processor. In the *Mule Debugger* view, inspect the content of the message as it exists at that point in the flow and evaluate a DataWeave expression against it.
```

Author: OK! Thanks!

> The debugger works with Enterprise Edition runtimes and doesn't connect to Mule Kernel runtimes.

Reviewer: Kernel init cap correct?

Author: According to [this entry](https://confluence.internal.salesforce.com/display/MTDT/Term+Presentation+Reference#:~:text=or%20%22Mule%20runtimes%22-,Mule%20Kernel,-Replaces%20Mule%20Community) in the [Term Presentation Reference](https://confluence.internal.salesforce.com/display/MTDT/Term+Presentation+Reference), yes.

To debug your application in Studio:

. Right-click your project in *Package Explorer*.
. Select *Debug As* > *Mule Application*.
+
Studio requests permission to switch to the *Mule debug* perspective.
. Click *Yes*.

> Additionally, run your application in debug mode by right-clicking in the canvas and selecting *Debug Project*, or by clicking the *Debug* button on the toolbar image:to-run-debug-mode-e7b7b.png[debug-button].

Reviewer: Do you mean "Alternatively"?

Author: You're right!

> Additionally, run your application in debug mode by right-clicking in the canvas and selecting *Debug Project*, or by clicking the *Debug* button on the toolbar image:to-run-debug-mode-e7b7b.png[debug-button].

Reviewer: ```suggestion
Alternatively, run your application in debug mode by right-clicking in the canvas and selecting *Debug Project*, or by clicking image:to-run-debug-mode-e7b7b.png[""] (*Debug*) on the toolbar.
```

Author: OK! Thanks!

== Configure Studio Debugger

Enable precise control over application debugging by setting the debugger connections and maximum values. To modify the default configuration values, click *Run* > *Debug configurations* and select the *Debug* tab.

[%header,%autowidth.spread,cols="a,a,a"]
|===
| Option | Description | Default Value

| Connection Properties 

> | Studio debugger listens to incoming TCP connections on `localhost` and port `6666`. Pause the debugger listening activities by selecting *Suspend* to prevent the debugger from triggering or processing any incoming requests temporarily. 

Reviewer: ```suggestion
| Studio debugger listens to incoming TCP connections on `localhost` and port `6666`. To temporarily prevent the debugger from triggering or processing incoming requests, click *Suspend* to pause debugger listening activities. 
```

Reviewer: Do you need the font changes here?

Author: So sorry, but I think I don't understand your comment :/

| 

* TCP connections: `localhost`

> * Port: `6666`

Reviewer: These bullets duplicate the first sentence in the description. Any way to address and make parallel with the other descriptions (imperative)?

Author: Let's see if my wording works.
